stdlib/REPL/src/REPLCompletions.jl

```diff
@@ -744,6 +770,76 @@ end
     return matches
 end
 
+# Identify an argument being completed in a method call. If the argument is empty, method
+# suggestions will be provided instead of argument completions.
+function identify_possible_method_completion(partial, last_idx)
+    fail = 0:-1, Expr(:nothing), 0:-1, 0
+
+    # First, check that the last punctuation is either ',', ';' or '('
+    idx_last_punct = something(findprev(x -> ispunct(x) && x != '_' && x != '!', partial, last_idx), 0)::Int
+    idx_last_punct == 0 && return fail
+    last_punct = partial[idx_last_punct]
+    last_punct == ',' || last_punct == ';' || last_punct == '(' || return fail
+
+    # Then, check that `last_punct` is only followed by an identifier or nothing
+    before_last_word_start = something(findprev(in(non_identifier_chars), partial, last_idx), 0)
+    before_last_word_start == 0 && return fail
+    all(isspace, @view partial[nextind(partial, idx_last_punct):before_last_word_start]) || return fail
+
+    # Check that `last_punct` is either the last '(' or placed after a previous '('
+    frange, method_name_end = find_start_brace(@view partial[1:idx_last_punct])
+    method_name_end ∈ frange || return fail
+
+    # Strip the preceding ! operators, if any, and close the expression with a ')'
+    s = replace(partial[frange], r"\G\!+([^=\(]+)" => s"\1"; count=1) * ')'
+    ex = Meta.parse(s, raise=false, depwarn=false)
+    isa(ex, Expr) || return fail
```

Comment:
Seems unfortunate we need all this code, since it seems like a slightly less complete version of https://github.com/JuliaLang/julia/blob/9bdaabdc829f7d26d8ab34cf422612161777f870/stdlib/REPL/src/REPLCompletions.jl#L905-L926
followed by a copy of
https://github.com/JuliaLang/julia/blob/9bdaabdc829f7d26d8ab34cf422612161777f870/stdlib/REPL/src/REPLCompletions.jl#L831-L834

Reply:
okay, I see the purpose is a bit different, since we are just looking for whether there is something quite trivial after the last punctuation, and the other code (and should_method_complete) is moved here now